src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.GenericDao;
+import core.basesyntax.factory.AbstractFactory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+
+    public static Object getInstance(Class clazz) throws IllegalAccessException,
+            InvocationTargetException, InstantiationException, NoSuchMethodException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] fields = clazz.getDeclaredFields();
+        for (Field field : fields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                if (field.getType().equals(GenericDao.class)) {
+                    field.set(instance, AbstractFactory.getNecessaryInstance(clazz));
```

Comment:
I can see that you want to make this look elegant
but it is not good to have two ConsoleHandlers, and that's why we need to refactor this

Reply:
done